test/hooks.test.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { expect } from 'chai';
+import 'mocha';
+import {
+    AppStartContext,
+    AppTerminateContext,
+    HookContext,
+    InvocationContext,
+    InvocationHookContext,
+    PostInvocationContext,
+    PreInvocationContext,
+    app,
+} from '../src/index';
+
+describe('hooks', () => {
+    it("register doesn't throw error in unit test mode", () => {
+        app.hook.appStart(() => {});
+        app.hook.appTerminate(() => {});
+        app.hook.postInvocation(() => {});
+        const registeredHook = app.hook.preInvocation(() => {});
+        registeredHook.dispose();
+    });
+
+    function validateHookContext(context: HookContext) {
```

Comment:
Having the helper functions in the middle of the `describe` block makes it a little harder to read through the tests. Maybe it might be easier to have them at the end of the block? We do something similar here: [test/errors.test.ts](https://github.com/Azure/azure-functions-nodejs-library/blob/f5af37479bb1327809756cf628456c33aed9e2ab/test/errors.test.ts)

Reply:
I don't think the rule is as simple as "helper functions at the end", but I see what you mean for this file. I like to have code close to where it's actually used so in some cases I like the helper functions in the middle.